src/libraries/Common/tests/System/Net/Http/GenericLoopbackServer.cs

```diff
@@ -184,7 +187,8 @@ public class HttpRequestData
         public string Path;
         public Version Version;
         public List<HttpHeaderData> Headers { get; }
-        public int RequestId;       // Generic request ID. Currently only used for HTTP/2 to hold StreamId.
+        public int RequestId;       // HTTP/2 StreamId.
+        public long Http3StreamId;
```

Comment:
I guess I'm really unhappy about this, but what about this: https://github.com/dotnet/runtime/compare/main...ManickaP:mapichov/h3_request_id?expand=1???

I can as well make it long, although I'm not sure we'll ever hit such large ID in our tests.

Reply:
We should definitely fix the current hack in GetRequestId() because that's just going to be super confusing if we keep it as it is.

That said, I looked at the usage of the RequestId field and it seems like it's not really used at all currently. Am I missing something here? If it's not used, let's just remove it and save ourselves the hassle.

For now, I can implement the GOAWAY support without having to add this field, which is probably the cleanest thing to do right now. I'll push an update.

Reply:
I found it used only here:
https://github.com/dotnet/runtime/blob/cef40a107ff2f810219b3b28af709b0b778a3472/src/libraries/Common/tests/System/Net/Http/Http2LoopbackConnection.cs#L664-L664

But that's within the same context as its assignment, so we can just use local variable instead.